hpx/util/spinlock_pool.hpp

```diff
@@ -41,20 +42,20 @@ namespace hpx { namespace util
     }
 #endif
 
-    template <typename Tag>
+    template <typename Tag, std::size_t N = 128>
     class spinlock_pool
```

Comment:
In theory, yes. In this implementation, no: The reason is the shift operation instead of division. We could manually clamp the value, but that would add another operation. I think requiring a power of two isn't that bad.